## Problem 2

In [1]:
ls -lh

total 2.4M
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  5 19:38  conv.txt
-rw-rw-r-- 1 osboxes osboxes  147 Sep  5 19:38  DISCLAIMER
-rw-rw-r-- 1 osboxes osboxes  505 Sep  5 19:51  Exam_1_Answers.ipynb
-rw-rw-r-- 1 osboxes osboxes 5.5K Sep  5 19:38 'Exam Week 1.ipynb'
-rw-rw-r-- 1 osboxes osboxes 3.9K Sep  5 19:38 'Exam Week 2.ipynb'
-rw-rw-r-- 1 osboxes osboxes 4.2K Sep  5 19:38  Germplasm.tsv
-rw-rw-r-- 1 osboxes osboxes  141 Sep  5 19:38  how_to_convert.txt
-rw-rw-r-- 1 osboxes osboxes 1.1K Sep  5 19:38  LICENSE
-rw-rw-r-- 1 osboxes osboxes  637 Sep  5 19:38  LocusGene.tsv
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  5 19:38  Locus_Germplasm_Phenotype_20130122.txt
-rw-rw-r-- 1 osboxes osboxes  743 Sep  5 19:38  README.md


Regarding the permissions of the files in the exams folder, the user (osboxes) can read and write all files (rw-). The group, in this case osboxes, can also read and write all files (rw-). Finally, 'everyone' can only read the files in this folder (r--)

## Problem 3

In [2]:
head -1 Locus_Germplasm_Phenotype_20130122.txt

Locus_name)	Germplasm_name	phenotype	pubmed_id


The command head with the flag -1 prints only the first line of the file 'Locus_Germplasm_Phenotype_20130122.txt'. That flag specifies the number of lines to print, but in this case we just want the first line.

## Problem 4

In [3]:
cat Locus_Germplasm_Phenotype_20130122.txt | wc -l

7216


To solve this problem we use the command cat to output to the stdout all the content of the file. Then we 'pipe' (|) the output of that command as input to the command wc (word count). As the word implies, this command counts the word of the input. We use the flag -l to specify that we only want the number of lines in the input. Thus we conclude that the file Locus_Germplasm_Phenotype_20130122.txt has 7216 lines.

## Problem 5